### PR TITLE
Add breadcrumbs to new product management experience

### LINKED ISSUE #35170: [Header] Breadcrumbs
**User story**

- As a merchant,
- I want to go back one or more steps in the navigation,
- So I can easily find the information or setting I am looking for.

**Design**
5sAIeTRd9Yp7nSCT33BAWz-fi-8924%3A388910

**Acceptance criteria**

- [ ] We show breadcrumbs in the left part of the new page header.
- [ ] Each breadcrumb is clickable.
- [ ] The arrows between the links are not clickable.
- [ ] The links get an underline on hover to indicate they're clickable.
- [ ] The current page's name (in this case, the product's name) is Gutenberg 900, bold, and not clickable.
- [ ] If there are more than 3 levels of breadcrumbs on the screen (excluding the currently viewed page), we truncate the path and only show the first and last steps. Example: `5sAIeTRd9Yp7nSCT33BAWz-fi-8924%3A388912`
- [ ] On mobile, the breadcrumbs are displayed above the current page's name and have different styling. Design: `5sAIeTRd9Yp7nSCT33BAWz-fi-7325%3A345827`.

### PR DESCRIPTION
**Pull request by joshuatf**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Adds breadcrumbs to the header of the new product management experience.

Screen Shot 2022-11-15 at 12 07 03 PM
Screen Shot 2022-11-15 at 12 05 27 PM

Closes #35170 .

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

1. Navigate to the new product management experience ("Products -> Add new (MVP)")
2. Note the breadcrumbs in the header next to the title
3. Currently, breadcrumbs should always be `Products > {product_name}` or `Products > New product`
4. Clicking on the breadcrumb should navigate to the product page
5. Hovering the breadcrumb should show an underline

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce-admin/client/products/product-breadcrumbs.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+					breadcrumbs[ breadcrumbs.length - 1 ],
+			  ]
+			: breadcrumbs;
+
+	return (
+		<span className="woocommerce-product-breadcrumbs">
+			{ visibleBreadcrumbs.map( ( breadcrumb ) => {
+				const { href, title, type } = breadcrumb;
+				return (
+					<>
```
Comment (mdperez86) [inline plugins/woocommerce-admin/client/products/product-breadcrumbs.tsx]:
```jsx 
<Fragment key={ key }>
```
Let's help React reconciliation in it's work ;)

**Review by mdperez86 — COMMENTED** [review]
Looks good. Just left a very small comment!

File: plugins/woocommerce-admin/client/products/product-breadcrumbs.tsx (hunk 1 above)
Comment (joshuatf) [reply plugins/woocommerce-admin/client/products/product-breadcrumbs.tsx]: Good call!  Just added in https://github.com/woocommerce/woocommerce/pull/35596/commits/ee4f0e8f97e23064f009f9253e15db447bd87a03

**Comment by joshuatf** [comment]
Thanks for the review, @mdperez86!  Updated per your feedback.

**Review by mdperez86 — APPROVED** [review]
Cool

**Pull request merged**